Assets/Character/Scripts/PlayerState/PlayerStateMove.cs

```diff
@@ -18,11 +20,27 @@ public override void Update()
         {
             ChangeState(PlayerStateEnum.TestJump);
         }
-		else if (PlayerInput.PressedRoll())
+        else if (roll_direction != 0)

```

Comment:
I really think you should wrap (GetButtonDown("Roll") && PressedMoveRight/Left) to get PressedRollRight() etc